source/extensions/filters/http/ext_authz/config.cc

```diff
@@ -52,13 +53,16 @@ Http::FilterFactoryCb ExtAuthzFilterConfig::createFilterFactoryFromProtoTyped(
 
     const uint32_t timeout_ms =
         PROTOBUF_GET_MS_OR_DEFAULT(proto_config.grpc_service(), timeout, DefaultTimeout);
-    auto async_client_cache = getAsyncClientCacheSingleton(context);
+    auto async_client_cache_singleton = Grpc::getAsyncClientCacheSingleton(context);
```

Comment:
Can we make the abstraction such that we don't have individual filters like ext_authz needing to reason about whether this is Envoy or Google gRPC any more? I think this is the end goal of this work. It's fine to punt to a future PR, I think myself or someone else in my team would pick this up this quarter.

Reply:
I investigated a little, and it's not very clear to me on how to implemented. I think we can work on it in a separate PR. As long as we have issues tracking it, it should be fine I think.